File: loadercomplex-forge/src/ver122/java/com/enderzombi102/loadercomplex/forge12/impl/utils/ForgeI18nSystem.java
package com.enderzombi102.loadercomplex.forge12.impl.utils;

import com.enderzombi102.loadercomplex.api.platform.I18nSystem;
import net.minecraft.client.resources.I18n;
import org.jetbrains.annotations.NotNull;

public class ForgeI18nSystem implements I18nSystem {
	@Override
	public @NotNull String translate( @NotNull String key ) {
		return I18n.format( key );
	}

	@Override
	public @NotNull String translate( @NotNull String key, Object... args ) {
		return I18n.format( key );
	}

	@Override
	public boolean contains( @NotNull String key ) {
		return I18n.hasKey( key );
	}

	@Override
	public @NotNull String getCurrentLang() {
		// TODO: Implement
		return "en_us";
	}
}

File: loadercomplex-forge/src/ver122/kotlin/com/enderzombi102/loadercomplex/forge122/impl/platform/ForgeI18nSystem.kt
package com.enderzombi102.loadercomplex.forge122.impl.platform

import com.enderzombi102.loadercomplex.api.platform.I18nSystem
import net.minecraft.client.resources.I18n

class ForgeI18nSystem : I18nSystem {
	override fun translate( key: String ): String =
		I18n.format( key )

	override fun translate( key: String, vararg args: Any ): String =
		I18n.format( key )

	override fun contains( key: String ): Boolean =
		I18n.hasKey( key )

	override fun getCurrentLang(): String = // TODO: Implement
		"en_us"
}
